src/utils/derive_ipfs_cid.ts

```diff
@@ -0,0 +1,9 @@
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const CID = require('ipfs-only-hash');
+
+type DataInput = string | Uint8Array | AsyncIterable<Uint8Array>;
```

Comment:
Consider type name `CIDHashable` - `DataInput` is wildly generic.

Reply:
f5f33ce4fdf7052d19cdf66f14fa55e45a7ed29b